crates/cdk-mint-rpc/src/proto/server.rs

```diff
@@ -0,0 +1,586 @@
+use std::net::SocketAddr;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::Arc;
+
+use cdk::mint::Mint;
+use cdk::nuts::nut04::MintMethodSettings;
+use cdk::nuts::nut05::MeltMethodSettings;
+use cdk::nuts::{CurrencyUnit, MintQuoteState, PaymentMethod};
+use cdk::types::QuoteTTL;
+use cdk::Amount;
+use thiserror::Error;
+use tokio::sync::Notify;
+use tokio::task::JoinHandle;
+use tokio::time::Duration;
+use tonic::transport::{Certificate, Identity, Server, ServerTlsConfig};
+use tonic::{Request, Response, Status};
+
+use crate::cdk_mint_server::{CdkMint, CdkMintServer};
+use crate::{
+    ContactInfo, GetInfoRequest, GetInfoResponse, RotateNextKeysetRequest,
+    RotateNextKeysetResponse, UpdateContactRequest, UpdateDescriptionRequest, UpdateIconUrlRequest,
+    UpdateMotdRequest, UpdateNameRequest, UpdateNut04QuoteRequest, UpdateNut04Request,
+    UpdateNut05Request, UpdateQuoteTtlRequest, UpdateResponse, UpdateUrlRequest,
+};
+
+/// Error
+#[derive(Debug, Error)]
+pub enum Error {
+    /// Parse error
+    #[error(transparent)]
+    Parse(#[from] std::net::AddrParseError),
+    /// Transport error
+    #[error(transparent)]
+    Transport(#[from] tonic::transport::Error),
+    /// Io error
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+}
+
+/// CDK Mint RPC Server
+#[derive(Clone)]
+pub struct MintRPCServer {
+    socket_addr: SocketAddr,
+    mint: Arc<Mint>,
+    shutdown: Arc<Notify>,
+    handle: Option<Arc<JoinHandle<Result<(), Error>>>>,
+}
+
+impl MintRPCServer {
+    pub fn new(addr: &str, port: u16, mint: Arc<Mint>) -> Result<Self, Error> {
+        Ok(Self {
+            socket_addr: format!("{addr}:{port}").parse()?,
+            mint,
+            shutdown: Arc::new(Notify::new()),
+            handle: None,
+        })
+    }
+
+    pub async fn start(&mut self, tls_dir: Option<PathBuf>) -> Result<(), Error> {
+        tracing::info!("Starting RPC server {}", self.socket_addr);
+
+        let server = match tls_dir {
+            Some(tls_dir) => {
+                tracing::info!("TLS configuration found, starting secure server");
+                let cert = std::fs::read_to_string(tls_dir.join("server.pem"))?;
+                let key = std::fs::read_to_string(tls_dir.join("server.key"))?;
+                let client_ca_cert = std::fs::read_to_string(tls_dir.join("ca.pem"))?;
+                let client_ca_cert = Certificate::from_pem(client_ca_cert);
+                let server_identity = Identity::from_pem(cert, key);
+                let tls_config = ServerTlsConfig::new()
+                    .identity(server_identity)
+                    .client_ca_root(client_ca_cert);
+
+                Server::builder()
+                    .tls_config(tls_config)?
+                    .add_service(CdkMintServer::new(self.clone()))
+            }
+            None => {
+                tracing::info!("No valid TLS configuration found, starting insecure server");
```

Comment:
```suggestion
                tracing::warn!("No valid TLS configuration found, starting insecure server");
```
IMO `warn` is better for the insecure branch.